core-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java

```diff
@@ -347,7 +348,18 @@ public void registerAdditionalActions(@NotNull DefaultActionGroup leftToolbar,
 
     leftToolbar.remove(manager.getAction(IdeActions.ACTION_RERUN));
     leftToolbar.remove(manager.getAction(IdeActions.ACTION_STOP_PROGRAM));
-    leftToolbar.remove(manager.getAction(IdeActions.ACTION_CLOSE));
+
+    // XDebugSessionTab puts this action second from end.
+    // Is there a predictable ID we could use instead? ACTION_CLOSE isn't it.
+    AnAction[] actions = leftToolbar.getChildActionsOrStubs();
```

Comment:
This does not appear to be working for me when I patch it in... actions[actions.length - 3] == pin tab, the correct value in my environment was actions[actions.length - 2]

I was digging around the code and the type of the action CloseAction is com.intellij.execution.ui.actions.CloseAction which as it turns out is never registered with the ActionManager for some reason, but clones its presentation from whatever action is IdeActions.ACTION_CLOSE

Could we just search through the actions list to find the type or text instead of guessing the position?